src/JWTAuthMiddleware.spec.ts

```diff
@@ -348,6 +348,38 @@ describe('JWTAuthMiddleware', () => {
           })
         )
       })
+      it('rejects if authorization is required and no authorization header is set', async () => {
+        const next = jest.fn()
+        const options = {
+          algorithm: EncryptionAlgorithms.HS256,
+          authorizationRequired: true,
+          secretOrPublicKey: 'secret'
+        }
+        await expect(
+          JWTAuthMiddleware(options).before(
+            {
+              callback: jest.fn(),
+              context: {} as any,
+              error: {} as Error,
+              // is there a way to make the headers optional based on the options passed to JWTAuthMiddleware?
```

Comment:
There would be a way, but this is actually a bug in the original code:
https://github.com/dbartholomae/middy-middleware-jwt-auth/blob/dc28c2d9356b532d76f70d60611f9c90fa54b7d5/src/JWTAuthMiddleware.ts#L71
In this line it should be `any` instead of `IAuthorizedEvent` because the function actually also handles events that are not formed like this.